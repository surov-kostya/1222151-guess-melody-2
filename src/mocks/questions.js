import {shape, string, oneOf, arrayOf, number} from 'prop-types';

export const questionType = shape({
  type: oneOf([`genre`, `artist`]).isRequired,
  genrge: oneOf([`rock`, `folk`, `jazz`, `pop`]),
  song: shape({
    src: string,
    artist: string
  }),
  answers: arrayOf(
      shape({
        id: number,
        src: string.isRequired,
        genrge: oneOf([`rock`, `folk`, `jazz`, `pop`]).isRequired
      })) |
    shape({
      id: number,
      picture: string.isRequired,
      artist: string.isRequired
    }).isRequired
});

export const settings = {
  gameTime: 5,
  errorCount: 3,
};

export const gameQuestions = [
  {
    type: `genre`,
    genre: `Hip Hop & Rap`,
    answers: [
      {
        id: 100,
        src: `https://www.youtube.com/audiolibrary_download?vid=3b32fa1674bc3764`,
        genre: `Hip Hop & Rap`,
      },
      {
        id: 101,
        src: `https://www.youtube.com/audiolibrary_download?vid=28782f0fee64fd45`,
        genre: `Dance & Electronic`,
      },
      {
        id: 102,
        src: `https://www.youtube.com/audiolibrary_download?vid=1b61f851e58e6ffe`,
        genre: `pop`,
      },
      {
        id: 103,
        src: `https://www.youtube.com/audiolibrary_download?vid=ad2e6ac237dfb8ba`,
        genre: `Dance & Electronic`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `Pop`,
    answers: [
      {
        id: 104,
        src: `https://www.youtube.com/audiolibrary_download?vid=1b61f851e58e6ffe`,
        genre: `Pop`,
      },
      {
        id: 105,
        src: `https://www.youtube.com/audiolibrary_download?vid=ad2e6ac237dfb8ba`,
        genre: `Dance & Electronic`,
      },
      {
        id: 106,
        src: `https://www.youtube.com/audiolibrary_download?vid=47be4e2187adf0c5`,
        genre: `Dance & Electronic`,
      },
      {
        id: 107,
        src: `https://www.youtube.com/audiolibrary_download?vid=fdce0df3f10e0a6f`,
        genre: `Ambient`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Bad Snacks`,
      src: `https://www.youtube.com/audiolibrary_download?vid=47be4e2187adf0c5`,
    },
    answers: [
      {
        id: 200,
        picture: `http://placehold.it/134x134`,
        artist: `Bad Snacks`,
      },
      {
        id: 201,
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        id: 202,
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Jeremy Blake`,
      src: `https://www.youtube.com/audiolibrary_download?vid=3b32fa1674bc3764`,
    },
    answers: [
      {
        id: 203,
        picture: `http://placehold.it/134x134`,
        artist: `Jeremy Blake`,
      },
      {
        id: 204,
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        id: 205,
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
];

export const songs = [
  {
    "title": `U In My Arms`,
    "download_url": `https://www.youtube.com/audiolibrary_download?vid=3b32fa1674bc3764`,
    "artist": `Jeremy Blake`,
    "genre": `Hip Hop & Rap`,
  },
  {
    "title": `Through The Crystal`,
    "download_url": `https://www.youtube.com/audiolibrary_download?vid=28782f0fee64fd45`,
    "artist": `Jeremy Blake`,
    "genre": `Dance & Electronic`,
  },
  {
    "title": `Everything Is Gonna Be Just Fine`,
    "download_url": `https://www.youtube.com/audiolibrary_download?vid=1b61f851e58e6ffe`,
    "artist": `Jeremy Blake`,
    "genre": `Pop`,
  },
  {
    "title": `Block Party`,
    "download_url": `https://www.youtube.com/audiolibrary_download?vid=ad2e6ac237dfb8ba`,
    "artist": `Bad Snacks`,
    "genre": `Dance & Electronic`,
  },
  {
    "title": `New Moon`,
    "download_url": `https://www.youtube.com/audiolibrary_download?vid=47be4e2187adf0c5`,
    "artist": `Bad Snacks`,
    "genre": `Dance & Electronic`,
  },
  {
    "title": `They Might Not`,
    "download_url": `https://www.youtube.com/audiolibrary_download?vid=fdce0df3f10e0a6f`,
    "artist": `Puddle of Infinity`,
    "genre": `Ambient`,
  }
];
