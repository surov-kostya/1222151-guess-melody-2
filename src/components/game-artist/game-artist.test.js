import React from 'react';
import renderer from 'react-test-renderer';
import {gameQuestions} from '../../mocks/questions';
import GameArtist from './game-artist';

it(`GameArtist renders correctly`, () => {
  const tree = renderer.create(<GameArtist question={gameQuestions[2]} screenIndex={2} onAnswer={() => { }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
