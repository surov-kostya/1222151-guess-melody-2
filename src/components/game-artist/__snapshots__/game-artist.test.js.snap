// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameArtist renders correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id={200}
          name="answer"
          onChange={[Function]}
          type="radio"
          value={200}
        />
        <label
          className="artist__name"
          htmlFor={200}
        >
          <img
            alt="Bad Snacks"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Bad Snacks
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id={201}
          name="answer"
          onChange={[Function]}
          type="radio"
          value={201}
        />
        <label
          className="artist__name"
          htmlFor={201}
        >
          <img
            alt="Jack Daniels"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Jack Daniels
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id={202}
          name="answer"
          onChange={[Function]}
          type="radio"
          value={202}
        />
        <label
          className="artist__name"
          htmlFor={202}
        >
          <img
            alt="Jim Beam"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Jim Beam
        </label>
      </div>
    </form>
  </section>
</section>
`;
